astropy/units/function/core.py

```diff
@@ -252,9 +253,18 @@ def to(self, other, value=1., equivalencies=[]):
             return self.function_unit.to(other_function_unit, value)
 
         else:
-            # when other is not a function unit
-            return self.physical_unit.to(other, self.to_physical(value),
-                                         equivalencies)
+            try:
+                # when other is not a function unit
+                return self.physical_unit.to(other, self.to_physical(value),
+                                             equivalencies)
+            except u.UnitConversionError as e:
+                if type(self) == u.MagUnit and self.function_unit == Unit('mag'):
+                    # One can get to raw magnitudes via math that strips the dimensions off.
+                    # Include extra information in the exception to remind users of this.
+                    msg = "Did you perhaps subtract magnitudes so the unit got lost?"
```

Comment:
There is another way to add a message that may be more visible:
```
e.args += (msg,)
raise e
```
It would give
"""
UnitConversionError: ("'' (dimensionless) and 'Jy' (spectral flux density) are not convertible", 'Did you perhaps subtract magnitudes so the unit got lost?')
"""

Possibly better might be:
```
e.args = (e.args[0] + "\n" + msg,)
raise e
```
which would give
"""
UnitConversionError: '' (dimensionless) and 'Jy' (spectral flux density) are not convertible
Did you perhaps subtract magnitudes so the unit got lost?
"""

Am also not 100% sure about it, though...